src/btree/bt_cursor.c

```diff
@@ -708,8 +701,14 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
      */
     if (valid) {
         exact = cbt->compare;
-        ret = __cursor_kv_return(cbt, cbt->upd_value);
-    } else if (__cursor_fix_implicit(btree, cbt)) {
+        WT_ERR(__cursor_kv_return(cbt, cbt->upd_value));
+        if (F_ISSET(cursor, WT_CURSTD_PREFIX_SEARCH))
+            valid = __wt_prefix_match(&state.key, &cbt->iface.key) == 0;
+        if (valid)
```

Comment:
Why not move `WT_ERR(__cursor_kv_return(cbt, cbt->upd_value));` to this if section?

Reply:
Can you add a comment?

Reply:
Is `__cursor_kv_return ` to retrieve the key to &cbt->iface.key so you have to call this before the prefix check? If so can you add a comment to describe this dependency before `__cursor_kv_return`. If not, why not move that call to the place as I suggested as in some cases, we don't need to do that.

Reply:
Can we only set the key and then set the value in the second valid check?

Reply:
Good points! I have reformatted the if and else statement to accommodate this. @quchenhao let me know if the logic looks good to you. By clearing the cursor I use `__cursor_state_restore` to restore the state of the cursor.